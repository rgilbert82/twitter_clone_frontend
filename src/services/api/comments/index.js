export { default as getCommentsAPI } from './getComments.js';
export { default as createCommentAPI } from './createComment.js';
export { default as editCommentAPI } from './editComment.js';
export { default as deleteCommentAPI } from './deleteComment.js';
