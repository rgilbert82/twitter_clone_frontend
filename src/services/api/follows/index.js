export { default as getFollowAPI } from './getFollow.js';
export { default as createFollowAPI } from './createFollow.js';
export { default as deleteFollowAPI } from './deleteFollow.js';
export { default as getUserFollowersAPI } from './getUserFollowers.js';
export { default as getUserFollowingsAPI } from './getUserFollowings.js';
