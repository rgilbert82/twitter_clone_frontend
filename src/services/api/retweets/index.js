export { default as createRetweetAPI } from './createRetweet.js';
export { default as deleteRetweetAPI } from './deleteRetweet.js';
