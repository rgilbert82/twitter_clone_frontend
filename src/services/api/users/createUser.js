import { get_API_URL } from '../../misc';

export default (userData) => {
  return new Promise((resolve, reject) => {
    const path = `${get_API_URL()}/users`;
    let request = new XMLHttpRequest();

    request.onreadystatechange = () => {
      if (request.readyState === XMLHttpRequest.DONE) {
        if (request.status === 200) {
          resolve(JSON.parse(request.responseText));
        } else {
          reject(request.responseText);
        }
      }
    }

    request.open('POST', path);
    request.setRequestHeader('Content-Type', 'application/json');
    request.send(JSON.stringify(userData));
  });
}
