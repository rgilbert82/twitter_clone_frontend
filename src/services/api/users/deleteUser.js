import { get_API_URL } from '../../misc';

export default (userId) => {
  return new Promise((resolve, reject) => {
    const path = `${get_API_URL()}/users/${userId}`;
    let request = new XMLHttpRequest();

    request.onreadystatechange = () => {
      if (request.readyState === XMLHttpRequest.DONE) {
        if (request.status === 200) {
          resolve(request.responseText);
        } else {
          reject(request.responseText);
        }
      }
    }

    request.open('DELETE', path);
    request.withCredentials = true;
    request.setRequestHeader('Authorization', document.cookie || '');
    request.send();
  });
}
