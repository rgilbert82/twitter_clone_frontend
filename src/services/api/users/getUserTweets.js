import { get_API_URL } from '../../misc';

export default (userID) => {
  return new Promise((resolve, reject) => {
    const path = `${get_API_URL()}/users/${userID}/tweets`;
    let request = new XMLHttpRequest();

    request.onreadystatechange = () => {
      if (request.readyState === XMLHttpRequest.DONE) {
        if (request.status === 200) {
          resolve(JSON.parse(request.responseText));
        } else {
          reject(request.responseText);
        }
      }
    };

    request.open('GET', path);
    request.withCredentials = true;
    request.setRequestHeader('Authorization', document.cookie || '');
    request.send();
  });
}
