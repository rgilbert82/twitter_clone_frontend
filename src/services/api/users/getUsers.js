import { get_API_URL } from '../../misc';

export default () => {
  return new Promise((resolve, reject) => {
    const path = `${get_API_URL()}/users`;
    let request = new XMLHttpRequest();

    request.onreadystatechange = () => {
      if (request.readyState === XMLHttpRequest.DONE) {
        if (request.status === 200) {
          resolve(JSON.parse(request.responseText));
        } else {
          reject(request.responseText);
        }
      }
    }

    request.open('GET', path);
    request.send();
  });
}
