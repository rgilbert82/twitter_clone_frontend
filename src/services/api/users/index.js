export { default as getUsersAPI } from './getUsers.js';
export { default as getUserAPI } from './getUser.js';
export { default as createUserAPI } from './createUser.js';
export { default as updateUserAPI } from './updateUser.js';
export { default as getUserTweetsAPI } from './getUserTweets.js';
export { default as deleteUserAPI } from './deleteUser.js';
