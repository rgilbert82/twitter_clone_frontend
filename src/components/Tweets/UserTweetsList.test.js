jest.mock('../../services/api/users');

import React from 'react';
import { UserTweetsList } from '.';
import { shallow } from 'enzyme';
import { getUserTweetsAPI } from '../../services/api/users';

describe(UserTweetsList, () => {
  const userID = 5;
  let fetchSpy = jest.spyOn(UserTweetsList.prototype, 'fetchTweets');
  let component;

  beforeEach(() => {
    component = shallow(<UserTweetsList user_id={userID} />);
  });

  afterAll(() => {
    fetchSpy.mockClear();
  });

  it("renders properly", () => {
    expect(component).toMatchSnapshot();
  });

  it("calls fetchTweets on mount", () => {
    expect(fetchSpy).toHaveBeenCalled();
  });

  it("fetches the tweets", () => {
    expect.assertions(2);
    component.instance().fetchTweets()
      .then(() => {
        expect(component.state('tweets')).toEqual([{ id: 1, body: 'something here', user_id: 5, image: null }]);
        expect(component.state('tweetsLoaded')).toEqual(true);
      });
  });
});
