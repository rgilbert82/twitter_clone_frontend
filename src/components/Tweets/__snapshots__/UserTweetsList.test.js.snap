// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class UserTweetsList extends _react2.default.Component {
  constructor(props) {
    super(props);
    this.state = {
      tweets: [],
      tweetsFeed: [],
      tweetsLoaded: false };

    this.fetchTweets = this.fetchTweets.bind(this);
    this.deleteTweet = this.deleteTweet.bind(this);
    this.deleteRetweet = this.deleteRetweet.bind(this);
    this.scrollListener = this.scrollListener.bind(this);
  }

  componentDidMount() {
    this._isMounted = true;

    if (this._isMounted) {
      this.fetchTweets();
      window.addEventListener('scroll', this.scrollListener);
    }
  }

  componentWillUnmount() {
    window.removeEventListener('scroll', this.scrollListener);
    this._isMounted = false;
  }

  componentDidUpdate() {
    const elements = document.getElementsByClassName('tweet_body_text');

    for (let i = 0; i < elements.length; i++) {
      elements[i].innerHTML = (0, _misc.hashtagify)(elements[i].innerHTML);
    }
  }

  scrollListener() {
    if (window.innerHeight + window.pageYOffset >= document.body.offsetHeight - 2) {
      if (this.state.tweetsFeed.length < this.state.tweets.length) {
        this.setState({ tweetsFeed: this.state.tweets.slice(0, this.state.tweetsFeed.length + 8) });
      }
    }
  }

  fetchTweets() {
    return (0, _users.getUserTweetsAPI)(this.props.user_id).
    then(data => {
      if (this._isMounted) {
        this.setState({ tweets: data, tweetsFeed: data.slice(0, 8), tweetsLoaded: true });
      }
    }).catch(() => {
      if (this._isMounted) {
        this.props.displayMessage('There was an error loading the tweets');
      }
    });
  }

  deleteTweet(tweetID) {
    return (0, _tweets.deleteTweetAPI)(tweetID).
    then(() => {
      this.setState({
        tweets: this.state.tweets.filter(tweet => {return tweet.id !== tweetID;}),
        tweetsFeed: this.state.tweetsFeed.filter(tweet => {return tweet.id !== tweetID;}) });

      this.props.lowerTweetCount();
    }).catch(() => {
      this.props.displayMessage('There was an error deleting the tweet');
    });
  }

  deleteRetweet(retweetID) {
    return (0, _retweets.deleteRetweetAPI)(retweetID).
    then(() => {
      this.setState({
        tweets: this.state.tweets.filter(tweet => {return tweet.retweet_id !== retweetID;}),
        tweetsFeed: this.state.tweetsFeed.filter(tweet => {return tweet.retweet_id !== retweetID;}) });

      this.props.lowerTweetCount();
    }).catch(() => {
      this.props.displayMessage('There was an error deleting the retweet');
    });
  }

  render() {
    let allTweets;

    if (this.state.tweetsLoaded) {
      allTweets = this.state.tweetsFeed.map(tweet => {
        let keyID;

        if (tweet.retweet) {
          keyID = (0, _misc.randomString)(5);
        } else {
          keyID = tweet.id;
        }

        return (
          _react2.default.createElement('li', { className: 'list_item', key: keyID, __source: { fileName: _jsxFileName, lineNumber: 106 }, __self: this },
            _react2.default.createElement(_.TweetListItem, { history: this.props.history, tweet: tweet, loggedIn: this.props.loggedIn, isCurrentUserPage: this.props.isCurrentUserPage, displayMessage: this.props.displayMessage, deleteTweet: this.deleteTweet, deleteRetweet: this.deleteRetweet, navigateToSearch: this.props.navigateToSearch, __source: { fileName: _jsxFileName, lineNumber: 107 }, __self: this })));


      });

      return (
        _react2.default.createElement('div', { __source: { fileName: _jsxFileName, lineNumber: 113 }, __self: this },
          _react2.default.createElement('h2', { className: 'sub_header', __source: { fileName: _jsxFileName, lineNumber: 114 }, __self: this }, 'Tweets'),
          _react2.default.createElement('ul', { __source: { fileName: _jsxFileName, lineNumber: 115 }, __self: this },
            allTweets)));



    } else {
      return _react2.default.createElement(_Misc.PageLoading, { __source: { fileName: _jsxFileName, lineNumber: 121 }, __self: this });
    }
  }} renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserTweetsList
    user_id={5}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
